vike/client/client-routing-runtime/prefetch/getPrefetchSettings.ts

```diff
@@ -67,7 +77,40 @@ function getPrefetchStaticAssets(pageContext: PageContextPrefetch, linkTag: HTML
   return 'hover'
 }
 
-function getPrefetchAttribute(linkTag: HTMLElement): PrefetchStaticAssets | null {
+function getPrefetchPageContext(pageContext: PageContextPrefetch, linkTag: HTMLElement): PrefetchPageContext {
+  {
+    const prefetchAttribute = getPrefetchPageContextAttribute(linkTag)
+    if (prefetchAttribute !== null) return prefetchAttribute
+  }
+
+  if ('prefetchPageContext' in pageContext.exports) {
+    const { prefetchPageContext } = pageContext.exports
+
+    const wrongUsageMsg = `prefetchPageContext should be an object with 'when' and 'expire' properties. 'when' should be false, 'hover', and 'expire' should be a number`
```

Comment:
Actually, thinking about it: how about we make the setting simply `boolean | number` instead?